commit 0bc0c4cfb2c2320314228c335f38b7a9a9b84c34
Author: Ulrich Drepper <drepper@redhat.com>
Date:   Wed Jun 25 23:25:56 2008 +0000

    (main): Rename to...
    (do_test): ... this. Remove cmdline option processing.
    (TIMEOUT): Define.
    (TEST_FUNCTION): Define.
    (CMDLINE_OPTIONS): Define.

---
 posix/tst-regex.c |   20 ++++++++++----------
 1 file changed, 10 insertions(+), 10 deletions(-)

--- a/posix/tst-regex.c
+++ b/posix/tst-regex.c
@@ -1,4 +1,4 @@
-/* Copyright (C) 2001, 2003 Free Software Foundation, Inc.
+/* Copyright (C) 2001, 2003, 2008 Free Software Foundation, Inc.
    This file is part of the GNU C Library.
 
    The GNU C Library is free software; you can redistribute it and/or
@@ -55,8 +55,8 @@
 			       size_t memlen, int icase, int expected);
 
 
-int
-main (int argc, char *argv[])
+static int
+do_test (void)
 {
   const char *file;
   int fd;
@@ -66,16 +66,9 @@
   char *outmem;
   size_t inlen;
   size_t outlen;
-  static const struct option options[] =
-    {
-      {"timing",no_argument,	&timing,	1 },
-      {NULL,	0,		NULL,		0 }
-    };
 
   mtrace ();
 
-  while (getopt_long (argc, argv, "", options, NULL) >= 0);
-
   /* Make the content of the file available in memory.  */
   file = "../ChangeLog.8";
   fd = open (file, O_RDONLY);
@@ -506,3 +499,10 @@
      expect.  */
   return cnt != expected;
 }
+
+/* If --timing is used we will need a larger timout.  */
+#define TIMEOUT 50
+#define CMDLINE_OPTIONS \
+   {"timing", no_argument, &timing, 1 },
+#define TEST_FUNCTION do_test ()
+#include "../test-skeleton.c"
